core/src/main/java/io/opencensus/stats/View.java

```diff
@@ -14,18 +14,24 @@
 package io.opencensus.stats;
 
 import com.google.auto.value.AutoValue;
-import io.opencensus.common.Function;
+import io.opencensus.common.Duration;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
 /**
  * A View specifies an aggregation and a set of tag keys. The aggregation will be broken
  * down by the unique set of matching tag values for each measure.
  */
 @Immutable
+@AutoValue
 public abstract class View {
+
+  View() {
+  }
+
   /**
    * Name of view. Must be unique.
    */
```

Comment:
Per our discussion today, I'll hold off changing `View.Name` until we have more details on naming constrains.